pytorch_lightning/plugins/checkpoint/checkpoint.py

```diff
@@ -0,0 +1,61 @@
+from abc import ABC
+from pathlib import Path
+from typing import Any, Dict, Mapping, Union
+
+from torch.nn import Module
+
+from pytorch_lightning import LightningModule
+
+
+class CheckpointPlugin(ABC):
+
+    def __init__(self):
+        self._training_type_plugin = None
+
+    @property
+    def training_type_plugin(self) -> 'TrainingTypePlugin':
+        return self._training_type_plugin
+
+    @training_type_plugin.setter
+    def training_type_plugin(self, plugin) -> None:
+        self._training_type_plugin = plugin
```

Comment:
Are we happy with the checkpoint plugin having a reference to the training type plugin? it's a bit cyclic but I think gives the freedom to define the checkpointing logic correctly.